### PR TITLE
test against Sequence not list

### LINKED ISSUE #7672: Ruamel Yaml CommentedSeq Subclasses MutableSequence not List
## Current Behavior
`conda` checks if something is a list, but since it is loaded by `ruamel_yaml` it is a `CommentedSeq`, which now no longer subclasses from `list` but from `MutableSequence`.

```python
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/conda/cli/main_config.py", line 30, in execute
    execute_config(args, parser)
  File "/opt/conda/lib/python3.6/site-packages/conda/cli/main_config.py", line 263, in execute_config
    raise CouldntParseError("key %r should be a list, not %s." % (key, bad))
conda.exceptions.CouldntParseError: key 'channels' should be a list, not CommentedSeq.
```
I think this is the relevant line in master. https://github.com/conda/conda/blob/master/conda/cli/main_config.py#L286

[Commit in `ruamel.yaml` changing inheritance](https://bitbucket.org/ruamel/yaml/commits/9e4de94ee6a88ed86ab5fdc6d49a23cfc6ab5782)

### Steps to Reproduce
Run a build with `ruamel_yaml>0.15.54`.

## Expected Behavior
Not error with `ruamel_yaml>0.15.54`

## Environment Information
Working on it.
<details open><summary><code>`conda info`</code></summary><p>

```

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```

```
</p></details>

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes #7672

### CODE REVIEW

File: conda/cli/main_config.py [hunk 1]
```diff
@@ -283,7 +290,8 @@ def execute_config(args, parser):
             if key not in sequence_parameters:
                 from ..exceptions import CondaValueError
                 raise CondaValueError("Key '%s' is not a known sequence parameter." % key)
-            if not isinstance(rc_config.get(key, []), list):
+            if not (isinstance(rc_config.get(key, []), Sequence) and not 
+                    isinstance(rc_config.get(key, []), str_type)):
```
Comment (mbargull) [inline conda/cli/main_config.py]: You can change `str_type` to `string_types` here.
`string_types` is already imported from `conda.common.compat` and does the same thing.

File: conda/cli/main_config.py [hunk 2]
```diff
@@ -19,6 +19,13 @@
 from ..common.io import timeout
 from ..common.serialize import yaml, yaml_dump, yaml_load
 
+try:
+    from collections.abc import Sequence, Mapping
+    str_type = str
+except ImportError:  # python 2
+    from collections import Sequence, Mapping
+    str_type = basestring
+
```
Comment (mbargull) [inline conda/cli/main_config.py]: I think we should put this into `conda.common.compat` and then do
```diff
-from collections import OrderedDict, Mapping
+from collections import OrderedDict
```
```diff
-from ..common.compat import isiterable, iteritems, itervalues, string_types
+from ..common.compat import Mapping, Sequence, isiterable, iteritems, itervalues, string_types
```
here.

**Comment by mbargull** [comment]
Oh, didn't notice this before, but can you retarget this at the `4.5.x` branch since `master` is currently targeted for the next feature release.

**Comment by mbargull** [comment]
I don't know why `tests/test_export.py::ExportIntegrationTests::test_multi_channel_explicit` failed. Since I'm curious, I went ahead with #7675 to target `4.5.x`.

**Comment by CJ-Wright** [comment]
Feel free to do whatever is needed for this patch.

**Comment by mbargull** [comment]
Thanks! The (relevant) tests went fine on the other PR, so we can use that one instead.

**Comment by kalefranz** [comment]
The `tests/test_export.py::ExportIntegrationTests::test_multi_channel_explicit` test failure has nothing to do with this PR. It's an integration test where something changed in metadata. Happens all the time. I've made the test more robust in another PR.

**Pull request closed without merging**